test/DnsClient.Tests/DatagramReaderTest.cs

```diff
@@ -54,11 +54,24 @@ public void DatagramReader_Labels_FromBytesValid()
             var bytes = new byte[] { 5, 90, 90, 92, 46, 90, 2, 56, 56, 0 };
 
             var reader = new DnsDatagramReader(new ArraySegment<byte>(bytes));
-
-            var labels = reader.ReadLabels();
+            var recursion = 0;
+            var labels = reader.ReadLabels(ref recursion);
             Assert.Equal(2, labels.Count);
         }
 
+        [Fact]
+        public void DatagramReader_Labels_FromBytesRecursionMax()
+        {
+            var bytes = new byte[] { 5, 90, 90, 92, 46, 90, 2, 56, 56, 0 };
+
+            var reader = new DnsDatagramReader(new ArraySegment<byte>(bytes), maxRecursion: 1);
```

Comment:
This is only testing the exception getting thrown, this is not actually testing if it prevents the StackOverflow exception (Which the current implementation does not do)